python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -7457,34 +7482,61 @@ def test_dataframe_sample_basic(n, frac, replace, axis):
             n=n,
             frac=frac,
             replace=replace,
-            random_state=random_state,
+            random_state=make_random_state(seed, random_state_lib),
             axis=axis,
         )
-        assert pout.shape == gout.shape
+        if random_state_lib == np.random.RandomState:
+            assert_eq(pout, gout)
+        else:
+            assert pout.shape == gout.shape
+
+
+@pytest.mark.parametrize("replace", [True, False])
+def test_dataframe_reproducibility_seed(replace):
+    df = cudf.DataFrame({"a": cupy.arange(0, 1024)})
+
+    seed = 1
+    expected = df.sample(1024, replace=replace, random_state=seed)
+    out = df.sample(1024, replace=replace, random_state=seed)
+
+    assert_eq(expected, out)
 
 
 @pytest.mark.parametrize("replace", [True, False])
-@pytest.mark.parametrize("random_state", [1, np.random.mtrand.RandomState(10)])
-def test_dataframe_reproducibility(replace, random_state):
+@pytest.mark.parametrize("lib", [np, cp])
```

Comment:
Could we just funnel into `make_random_state` instead of reimplementing the logic here? Then if we add other types of state etc we would only need to change the parametrization, not the logic of the test.